### PR TITLE
Docker args config

### PR DESCRIPTION
**Pull request by bigzachattack**
Support both create_container and start optional args. Made args configurable, so you do not have to extend DockerSpawner to just change the args.

### CODE REVIEW

File: dockerspawner/dockerspawner.py [hunk 1]
```diff
@@ -216,7 +218,7 @@ def get_container(self):
         self.container_id = ''
     
     @gen.coroutine
-    def start(self, image=None, **extra_create_kwargs):
+    def start(self, image=None):
```
Comment (jhamrick) [inline dockerspawner/dockerspawner.py]: I think we had talked about making it both an instance-level config option, and something that could still be passed in here by subclasses (e.g. in the case when you need per-container configuration, like [setting the working directory to be the home directory](https://github.com/compmodels/jupyterhub-deploy/blob/master/roles/jupyterhub_host/files/swarmspawner.py#L49)).

File: dockerspawner/dockerspawner.py [hunk 2]
```diff
@@ -250,12 +252,17 @@ def start(self, image=None, **extra_create_kwargs):
...
+        # build the dictionary of keyword arguments for start
+        start_kwargs = dict(
             binds=self.volume_binds,
-            port_bindings={8888: (self.container_ip,)}
-        )
+            port_bindings={8888: (self.container_ip,)})
+        start_kwargs.update(self.extra_start_kwargs)
+
+        # start the container
+        resp = yield self.docker('start', self.container_id, **start_kwargs)
```
Comment (jhamrick) [inline dockerspawner/dockerspawner.py]: I don't think this `resp` is being used for anything?

**Comment by bigzachattack** [comment]
@jhamrick Would ```def start(self, image=None, extra_create_kwargs={}, extra_start_kwargs={})``` work?

**Comment by jhamrick** [comment]
Thanks! I would just say one other change -- could you make it be `extra_create_kwargs=None` and `extra_start_kwargs=None` in the function definition, rather than `{}`? There can be some nasty unpredictable bugs if you make default keyword objects be mutable objects. So if you set them to `None`, then you can check in the function body whether they're `None`, and if so, set them to `{}` there.

**Comment by bigzachattack** [comment]
Thanks!

**Comment by jhamrick** [comment]
This looks great! I actually just realized, though, that this change will also require updating `systemuserspawner.py` because it still assumes the `**kwargs` function signature: https://github.com/jupyter/dockerspawner/blob/master/dockerspawner/systemuserspawner.py#L119

Could you update that, too, so that it passes the working directory as part of `extra_create_kwargs`?

**Comment by bigzachattack** [comment]
Any other place where this will trickle down that I need to go clean up?

**Comment by jhamrick** [comment]
Perfect. I think those are all the places we need to worry about in this repository, at least.

**Pull request merged**